### DISCUSSION TITLE
Sneak preview - Update on the working in progress

### DISCUSSION THREAD

**Discussion by urania-dev** [urania-dev/snapp#15] [General]
Just to let you see how is coming on :)

![Screenshot 2024-01-26 at 19-07-32 https __labs snapp li](https://github.com/urania-dev/snapp/assets/153094889/2b5dcdce-fdde-491b-a5ea-6f3ec0c2ddb2)
![Screenshot 2024-01-26 at 19-07-23 https __labs snapp li](https://github.com/urania-dev/snapp/assets/153094889/9744f7c5-f679-47b0-8969-29072461f086)
![Screenshot 2024-01-26 at 19-07-07 https __labs snapp li](https://github.com/urania-dev/snapp/assets/153094889/62e23e35-8b24-4844-acfe-625092590633)
![Screenshot 2024-01-26 at 19-06-44 https __labs snapp li](https://github.com/urania-dev/snapp/assets/153094889/a23da09d-29ca-47e5-b3a3-845c49348e14)
![Screenshot 2024-01-26 at 19-14-20 https __labs snapp li](https://github.com/urania-dev/snapp/assets/153094889/1f064cad-e16b-4026-b7f7-6b072e67a7ea)

**Comment by Hutch79** [comment]
Uhh, seeing some features in there that are new!
I'm unsure about the new Boxy look, preferred the slightly rounded edges of the old design.
It's somehow strange to look at a gray Snapp instead of a blue one ^^

Curious about future sneak peaks and ultimately the update!

**Reply by urania-dev** [reply]
![Screenshot 2024-01-26 at 20-41-04 https __labs snapp li](https://github.com/urania-dev/snapp/assets/153094889/a3ca6a5c-34ff-49a1-b477-e8e2d14d0b66)

well, i should be able to allow for customization with [https://www.skeleton.dev/docs/generator](https://www.skeleton.dev/docs/generator)

**Reply by Hutch79** [reply]
> well, i should be able to allow for customization with https://www.skeleton.dev/docs/generator
🥳 🎈 🎉

**Comment by urania-dev** [comment]
It's almost done, i'm ok with launching a 0.7.test version on docker hub, to let people test and debug, since the rebuild with all the new features kind of confused me at some point. 

ergo, it should work, it's not fully tested, hence my proposal, 

still missing the new homepage.

**Reply by Hutch79** [reply]
I would be happy to give it a look!
If I find any Bugs, I'll let you know.

Is the migration from 0.6.1 automatic, or do I need to import Snapps via csv?

**Reply by urania-dev** [reply]
At the moment there is no automation, but i think i'll put a CSV import and by now the redis panel should already include some sort of it. 

From test to release i aim to address all requests. 

If you want a preview i left open the test hosted on my homelab, you can reach it at labs.snapp and it should allows registration. Tomorrow i'll deploy the test image for own setups.

Also the code was written starting from zero all over, so i think i will branch the repo for the legacy version 🤔

**Reply by Hutch79** [reply]
> Also the code was written starting from zero all over, so i think i will branch the repo for the legacy version 🤔
Sounds logical.
Currently doing the same with one of my projects.

> Tomorrow i'll deploy the test image for own setups.
💜

**Reply by Hutch79** [reply]
Erm....
![image](https://github.com/urania-dev/snapp/assets/42042811/f2f27966-1d7a-4747-8f80-3aecdfcb6bf2)

I think I found a bug ^^
If I add a special character (`!`) to it, it works

**Reply by urania-dev** [reply]
Yeah, It takes times but i never learned this much alltogether. 

By the way: expect bugs, a lot.

 I just noticed the mobile version has a bug in language switcher 😅

Also around the limits and new features. They need to be tested

**Reply by Hutch79** [reply]
For bugs, Should I report them here or in a new discussion/issue?

**Reply by urania-dev** [reply]
> Erm....
> ![image](https://github.com/urania-dev/snapp/assets/42042811/f2f27966-1d7a-4747-8f80-3aecdfcb6bf2)
> 
> I think I found a bug ^^
> If I add a special character (`!`) to it, it works
> 
> 

Sorry, still sleepy and I edited your post instead of reply 😴

Yeah copy needs an upgrade, i changed the regex to a more stable and secure pattern 😅

**Reply by urania-dev** [reply]
> For bugs, Should I report them here or in a new discussion/issue?

Leave them here for now. Until an official test release, where i'll create a dedicated thread

**Reply by Hutch79** [reply]
If I edit a Snapp and want to save it, it tells me I've reached the max number of Snapps I'm allowed to have.
I am able to create new Snapps.

**Reply by urania-dev** [reply]
Mmmh. Interesting. The script is very similar but i think i know where it gets stuck. 

I'll try to fix everything you find before test release.

Also, mind that mondays i have a double shift at work, so for today i'll only be able to read comments, but i really appreciate your help! 

Thank you!

How do you find the new ui? :)

**Reply by Hutch79** [reply]
Requirements: A Snapp with Password protection.

If somebody opens this Snapp, he's able to enter the password, but neither the Button nor the Enter key lets him proceed.
Also, the password is displayed in plain text.

**Reply by Hutch79** [reply]
> Also, mind that mondays i have a double shift at work, so for today i'll only be able to read comments, but i really appreciate your help!

Hey, no problem!
You're one of the most active devs I've ever seen ^^
I know of other resources which are great, but support is basically non-existent...

**Reply by urania-dev** [reply]
Perfect. Yeah some pages i've worked on nighttime.  I'll fix those 🙈

**Reply by Hutch79** [reply]
> How do you find the new ui? :)

I must admit, I prefer the old one.
It seems more harmonic with Its colors
The contrast between background and foreground in the new version is a bit too much for my taste ^^

But other stuff like the Notes (btw. where are they displayed?) and that the Snapp link next to the checkbox on the Dashboard doesn't open the Snapp anymore and instead redirects to the stats page is great!

I also like that the Stats now can be seen on a per Snapp basis, although a general view with all Snapps a user has would be nice as well ^^
But I think the old Stats page was a bit better. It needed more space, but it displayed everything and didn't hide it behind a dropdown.

Also, little things like setting an individual page size for pagination are the icing on a cake!

And of course Translations! ^^

**Comment by urania-dev** [comment]
> must admit, I prefer the old one. It seems more harmonic with Its colors The contrast between background and foreground in the new version is a bit too much for my taste ^^

i think we can work together to generate a new theme that's easy on the eye and that will garantee accessibility, the use of skeletondev actually integrate a lot of accessibility and aria options that are great at that. so it is just a term of deciding a palette we enjoy. 

also i'll leave the option to integrate a css stylesheet to override the theme atomic variables, so customization will be effective on an instance basis, without passing by the code or rebuilding the app.

> But other stuff like the Notes (btw. where are they displayed?) and that the Snapp link next to the checkbox on the Dashboard doesn't open the Snapp anymore and instead redirects to the stats page is great!

yeah they should appear in the snapp single page. beneath the data of the single snapp, still a work in progress.

> I also like that the Stats now can be seen on a per Snapp basis, although a general view with all Snapps a user has would be nice as well ^^ But I think the old Stats page was a bit better. It needed more space, but it displayed everything and didn't hide it behind a dropdown.

i tought that ease that page load of information would have been a good idea, but it seems i was mistaken, i can split the lists behind the dropdown in the general metrics no prob. a little old version style with improvements. has to be honest about it=> the map yet doesn't get updated, but this time you can save prefereed position on the wolrd map :D

> all Snapps a user has would be nice as well ^^

it should be already there, what am i missing? 
- a dashboard with all the users (admin) has created /dashboard 
- the snapps of any user /dashboard/users/[username] 
- the single snapp metrics /dashboard/urls/[shortcode]
- the whole site metrics /dashboard/metrics
🤔 

 
> Also, little things like setting an individual page size for pagination are the icing on a cake!

:)

**Reply by urania-dev** [reply]
btw. setting you as an admin so you can lurk around, found another bug: while editing your profile i HAD to put an email, so your new profile email is info@example.com, just so you know ^^"

**Reply by urania-dev** [reply]
> And of course Translations! ^^

my suggestion is to take batch from the file to put in through chatgpt, with the italian version it worked. :D

**Reply by Hutch79** [reply]
> it should be already there, what am i missing?

Only there if you're Admin. So as User there is no metrics tab.
Also, as Admin you'll see all Snapp stats, not just yours.
___
> also i'll leave the option to integrate a css stylesheet to override the theme atomic variables, so customization will be effective on an instance basis, without passing by the code or rebuilding the app.

Uhh, Nice!
Snapp themes can go wild then ^^
___
> my suggestion is to take batch from the file to put in through chatgpt, with the italian version it worked. :D

Our dear old friend ChatGPT ^^
I would suggest a mixture between AI or other translation tools and actually look at it ^^
I noticed that GPT sometimes completely failed when I tried to translate something.
Tolgee would be a tool which allows for translating in a controlled manner. It also supports integration with DeepL, Google Translate and others.

**Reply by Hutch79** [reply]
New Bug:
If you try to create a new User as Admin, you'll prompted to enter a password, but there's no field for it.

**Reply by Hutch79** [reply]
Bug:
In the User management, if i click on a user to see his/here Snapps, I only see an empty list with the checkbox and the 3 dots menu but no additional info.
![image](https://github.com/urania-dev/snapp/assets/42042811/f352f1dc-19ca-452f-b5ba-78c8dfe65e49)

**Reply by Hutch79** [reply]
Btw. Where is the logout button? XD

**Reply by Hutch79** [reply]
Bug, maybe:
/privacy-policy does show a 404 Error. No Snapp Found with this name.
Not sure if this is a bug or not ^^
Could be intentional to set a policy via Snapp?

I tested it and the "privacy policy" is now online XP

**Reply by urania-dev** [reply]
> btw. Where is the logout button? XD

on the homepage, as a placeholder for the text i need to yet put on it 🗡️

i know i know, it's hardly a complete job. but it is a good point to start testing, as the features are many...

> /privacy-policy 
that is just because the page doesn't exists, yet. but you found the feature behind the bug 🤣

**Reply by urania-dev** [reply]
that is because the columns are deactivated, i forgot to place a default on that. you can save you preference and will keep it :)

**Reply by Hutch79** [reply]
I think the privacy policy link should stay the way it is ^^
Let the user decide where to redirect others XD

**Reply by urania-dev** [reply]
Fixed some stuff here and there. 
- Snapp code is now hidden. Also now should work with button and enter. 
- metrics now looks a little more like the old version, and Map get filled with data 
- metrics available for user, filtered if not admin, whole snapp metrics if admin
- notes now display on single shortcode page
- also seems to me a little less heavy on the browser, i'll look into it, cause metrics page seems heavier than this now 
- language switch fixed for mobile
- set default colums for user table
- fixed useless limit check on edit 


Would love to test the api limits. But besides click-spamming postman and hope it gets stuck i don't know

**Reply by urania-dev** [reply]
still to check: 
- create a new User as Admin, you'll prompted to enter a password, but there's no field for it.
- TTL seems to get placeholder as value even if off, fixable removing placeholder on toggle off 
- email placeholder in search field
- check email mandatory on emailess account  edit
 > Tolgee would be a tool which allows for translating in a controlled manner. It also supports integration with DeepL, Google Translate and others 

A little overkill for me, im unable to do that. To be honest i passed It with cgpt and then corrected the Italian typo here and there where occurred. Took 20 Min to divide and double check the keys for random (luckly only three on a 300keys file)


Also tried a less contrasty theme look and feel. Let me know if it is better :)

**Reply by Hutch79** [reply]
> Also tried a less contrasty theme look and feel. Let me know if it is better :)

I heard green is the new blue? XD
Looks pretty good to me!
Maybe make the text of the "New Snapp" button fat. It's kinda hard to read from far.
___
Btw. is it normal that I lost all my Snapps?
I guess It's because Alpha.
___
New Bug:
When entering the Password on a protected Link, the main enter button works, but the one on the Num Block doesn't.
___
For API testing, a Friend recommended [Postman](https://www.postman.com/).
___
Where can I enable a disabled snapp again?
___
> Took 20 Min to divide and double check the keys for random (luckly only three on a 300keys file)

Yea, ok, that's faster ^^

**Reply by urania-dev** [reply]
No you lost them cause as another tester found if you leave without expirations it gets the placeholder and make them expire on Midnight 😅

Will change new snapp button, no problem. 

Lost to the love of their families: signout button, disabile snapp button 😅

**Reply by Hutch79** [reply]
> No you lost them cause as another tester found if you leave without expirations it gets the placeholder and make them expire on Midnight 😅

I would call this an unintended Demo mode XD

**Reply by urania-dev** [reply]
Well on a bright side, expiration works😂

**Comment by urania-dev** [comment]
we should be close now. 

last round of bugs and then i'll release the 0.7 with migrations and stuff :)

**Reply by Hutch79** [reply]
Awesome!
I'm on holidays next week, so I unfortunately can't really test it.

Also... I must admit that I somehow messed up my admin login on labs.snapp ^^'
Could you change the mail address to the one of 2hutch79? 
I'll take it as an opportunity to test the forgot password process XD

**Reply by urania-dev** [reply]
Actually, i removed them while trying SMTP features. 

I'll put It back on!

Also, previous bug fixed remove all emails on language change 😂

**Reply by urania-dev** [reply]
Uploaded the almost complete version with importer and fixes. 

It should be missing just the home
